src/formatDate.js

```diff
@@ -8,7 +8,48 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
```

Comment:
Code quality issue: the `formatDate` function is a bit too long and does a lot of things. It would be better to split it into smaller functions, each doing one specific thing. For example, you could create a separate function for splitting and reordering the date.